bitcoin/src/crypto/key.rs

```diff
@@ -220,12 +220,8 @@ pub struct SortKey(ArrayVec<u8, 65>);
 
 impl fmt::Display for PublicKey {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        // TODO: fast hex encoding
         self.with_serialized(|bytes| {
-            for ch in bytes {
-                write!(f, "{:02x}", ch)?;
-            }
-            Ok(())
+            write!(f, "{}", bytes.as_hex())
```

Comment:
This drops formatting options, I think you should just delegate. And yes, formatting wasn't available before but let's add it since it's trivial.

Reply:
I appreciate getting my code reveiwed by you @Kixunil, I hope that one day I learn to notice all the things that you notice.